### PR TITLE
feat: new go template

### PR DESCRIPTION
**Pull request by anandsunderraman**
**Description**

New Go template to generate a go module with watermill.io

This PR is meant to replace the PR https://github.com/asyncapi/go-template/pull/23

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,24 +1,110 @@
...
+        "@semantic-release/release-notes-generator",
+        {
+          "preset": "conventionalcommits"
+        }
+      ],
+      "@semantic-release/npm",
+      "@semantic-release/github"
+    ]
   },
   "generator": {
```
Comment (derberg) [inline package.json:63]: please add info about supported protocol -> https://github.com/asyncapi/generator/blob/master/docs/authoring.md#configuration-file

**Review by derberg — COMMENTED** [review]
you should not remove GH Action workflow, any reason you did it?

File: readme.md [hunk 2]
```diff
@@ -1,161 +1,112 @@
...
-```
\ No newline at end of file
+5. Running local instance of `rabbitmq`, navigate to it using `http://localhost:15672/` with username and password `guest`/ `guest` (These are default rabbitmq credentials). 
+   FYI one can start an instance of `rabbitmq` using  `docker` as follow
+   ```
+   docker run -d -p 15672:15672 -p 5672:5672 rabbitmq:3-management
+   ```
+6. Create a queue as per the AsyncAPI spec. 
+   This can be done either of the following ways
+   a. Using the UI: Refer to this [article](https://www.cloudamqp.com/blog/part3-rabbitmq-for-beginners_the-management-interface.html) that walks through the process of how this can be done in the UI / RabbitMQ Admin
```
Comment (smoya) [inline readme.md:72]: This multilevel list won't render properly. Rather you should use something like:

```
1. One
   - One.1
   - One.2
2. Two
   - Two.1
   - Two.2
3. Three
   - Three.1
   - Three.2
```

File: readme.md [hunk 3]
```diff
@@ -1,161 +1,112 @@
...
+8. Check the output at the terminal where `go run main.go` was running and the published message should be printed
+
+## Template configuration
+
+You can configure this template by passing different parameters in the Generator CLI: `-p PARAM1_NAME=PARAM1_VALUE -p PARAM2_NAME=PARAM2_VALUE`
+
+|Name|Description|Required|Example|
+|---|---|---|---|
+|moduleName|Name for the generated Go module|false|`my-app`|
+
```
Comment (smoya) [inline readme.md:108]:
```suggestion
```

File: readme.md [hunk 4]
```diff
@@ -1,161 +1,112 @@
...
+
+
+
+
```
Comment (smoya) [inline readme.md:112]:
```suggestion
```

File: readme.md [hunk 5]
```diff
@@ -1,161 +1,112 @@
...
+      -H 'cache-control: no-cache' \
+      -H 'content-type: application/json' \
+      -d ' {
+    "properties":{},
+    "routing_key":"light/measured",
+    "payload":"{\"id\":1,\"lumens\":2,\"sentAt\":\"2021-09-21\"}",
+    "payload_encoding":"string"
+    }'
+   ```
+8. Check the output at the terminal where `go run main.go` was running and the published message should be printed
```
Comment (smoya) [inline readme.md:99]: Nitpick: 
```suggestion
8. Check the output at the terminal where `go run main.go` was running and the published message should be printed.
```

File: package.json (hunk 1 above)
Comment (anandsunderraman) [reply package.json:63]: makes sense will do

File: readme.md (hunk 2 above)
Comment (anandsunderraman) [reply readme.md]: done makes sense

File: readme.md (hunk 4 above)
Comment (anandsunderraman) [reply readme.md]: thank you done

**Comment by anandsunderraman** [comment]
@derberg I added the missing github workflows and a few other files I missed out. Also finally managed to rename the readme.md. Phew never knew it was not straight forward

File: .github/workflows/automerge-orphans.yml [hunk 6]
```diff
@@ -20,13 +20,14 @@ jobs:
           script: |
             const query = `query($owner:String!, $name:String!) {
               repository(owner:$owner, name:$name){
-                pullRequests(first: 100, states: OPEN){
+                pullRequests(first: 100){
```
Comment (derberg) [inline .github/workflows/automerge-orphans.yml:23]: this file must entirely reflect https://github.com/asyncapi/.github/blob/master/.github/workflows/automerge-orphans.yml

File: .github/workflows/automerge.yml [hunk 7]
```diff
@@ -26,7 +26,7 @@ jobs:
     steps:
       - name: Autoapproving
         uses: hmarr/auto-approve-action@v2
-        if: github.actor == ('asyncapi-bot' || github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]') && !contains(github.event.pull_request.labels.*.name, 'released')
+        if: github.actor == 'asyncapi-bot' || github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
```
Comment (derberg) [inline .github/workflows/automerge.yml:29]: please reflect this file in full https://github.com/asyncapi/.github/blob/master/.github/workflows/automerge.yml

File: .github/workflows/issues-prs-notifications.yml [hunk 8]
```diff
@@ -12,9 +10,6 @@ on:
   pull_request_target:
     types: [opened, reopened, ready_for_review]
 
-  discussion:
```
Comment (derberg) [inline .github/workflows/issues-prs-notifications.yml:15]: same as with other ones https://github.com/asyncapi/.github/blob/master/.github/workflows/issues-prs-notifications.yml

File: .github/workflows/welcome-first-time-contrib.yml [hunk 9]
```diff
@@ -1,34 +1,30 @@
 #This action is centrally managed in https://github.com/asyncapi/.github/
 #Don't make changes to this file in this repo as they will be overwritten with changes made to the same file in above mentioned repo
 
-#########
-#disabled because of https://github.com/asyncapi/.github/issues/73
-#########
+name: Welcome first time contributors
```
Comment (derberg) [inline .github/workflows/welcome-first-time-contrib.yml:4]: must be disabled, please replace with https://github.com/asyncapi/.github/blob/master/.github/workflows/welcome-first-time-contrib.yml

**Review by derberg — COMMENTED** [review]
just some fixes in workflows, and then I think we are ready to merge and do initial release

**Comment by derberg** [comment]
@anandsunderraman another important topic is the CODEOWNER-ship of the project. I know you for a while already, you started working on it some time ago and it wasn't easy as we were pushing first for reuse of existing work. You presented strong motivation and did not run away like others but actually tried to push topic forward and here we are, PR is ready to merge.

Can I assume you want to maintain this code generator in the future? I hope so 😄 if yes, than please:
- add such a file and specify your github handle there, only yours https://github.com/asyncapi/cupid/blob/master/CODEOWNERS
- we do not have to many Go experts in the project contributing regularly, so I suggest that maybe @smoya can join here as a CODEOWNER? you already did great job in the initial review. Up to you folks to decide really
- @anandsunderraman once you are ready to maintain it long term please consider adding yourself to the list of [AsyncAPI TSC members](https://github.com/asyncapi/community/blob/master/TSC_MEMBERS.json) and familiarize with [what it means](https://github.com/asyncapi/community/blob/master/CHARTER.md#2-technical-steering-committee) (much more [human-friendly text](https://www.asyncapi.com/blog/governance-motivation))

File: package.json [hunk 10]
```diff
@@ -1,24 +1,107 @@
...
   "main": "index.js",
   "scripts": {
-    "test": "echo \"no test specified yet\"",
-    "lint": "echo \"no lint configured yet\"",
-    "generate:assets": "echo \"no assets to generate yet\"",
-    "develop": "ag test/asyncapi.yaml ./ -o test/output --force-write --watch-template"
+    "test": "npm run testTemplate && npm run testHooks && npm run testFilters",
+    "testTemplate": "rimraf test/temp && jest --coverage -t 'templateGenerationResult' --passWithNoTests",
+    "testHooks": "rimraf test/temp && jest --coverage -t 'generateExtraFormats' --passWithNoTests",
+    "testFilters": "rimraf test/temp && jest --coverage -t 'generateMermaidDiagram' --passWithNoTests",
```
Comment (derberg) [inline package.json:10]: I don't think you have tests here in the project yet, so just keep `test` script that echos there are no tests. And as a followup create an issue to add snapshot tests that are a kind of integration tests checking if generated output is what you expected. I can later introduce you to it, not a problem at all

**Comment by smoya** [comment]
> * we do not have to many Go experts in the project contributing regularly, so I suggest that maybe @smoya can join here as a CODEOWNER? you already did great job in the initial review. Up to you folks to decide really

I'd be happy to be a code owner :)

File: .github/workflows/automerge-orphans.yml (hunk 6 above)
Comment (anandsunderraman) [reply .github/workflows/automerge-orphans.yml:23]: thanks will do

File: .github/workflows/automerge.yml (hunk 7 above)
Comment (anandsunderraman) [reply .github/workflows/automerge.yml:29]: ok thanks will do

File: .github/workflows/issues-prs-notifications.yml (hunk 8 above)
Comment (anandsunderraman) [reply .github/workflows/issues-prs-notifications.yml:15]: ok will do

File: .github/workflows/welcome-first-time-contrib.yml (hunk 9 above)
Comment (anandsunderraman) [reply .github/workflows/welcome-first-time-contrib.yml:4]: ok will do

File: package.json (hunk 10 above)
Comment (anandsunderraman) [reply package.json]: ok done, make sense

**Comment by anandsunderraman** [comment]
@derberg @smoya thanks for all your help and patience in reviewing this. 
I am happy to help maintain this repo along with @smoya going forward.
Open to any feedback !!

**Comment by derberg** [comment]
@smoya thank you!
@anandsunderraman so happy to hear that, thanks 🙇🏼 please add CODEOWNER file to this PR with you and @smoya on the list (and bot 😄 )

**Comment by anandsunderraman** [comment]
@derberg added the CODEOWNER to this PR https://github.com/asyncapi/go-template/pull/25/files#diff-fcf14c4b7b34fe7a11916195871ae66a59be87a395f28db73e345ebdc828085b

**Review by derberg — APPROVED** [review]
Let's :shipit: 
@anandsunderraman once again 💯 thanks for your patience and persistence in getting it done! huge respect from my side!!!

@smoya can you approve too senior?

**Review by smoya — APPROVED** [review]
🚀

**Pull request merged**